src/mlpack/core/metrics/non_maximal_supression_impl.hpp

```diff
@@ -0,0 +1,126 @@
+
+  // Clear selected bounding boxes.
+  selectedBoundingBoxes.clear();
+
+  // Obtain Sorted indices for bounding boxes according to
+  // their confidence scores.
+  arma::ucolvec sortedIndices = arma::sort_index(confidenceScores);
+
+  // Pre-Compute area of each bounding box.
+  arma::mat area;
+  if (UseCoordinates)
+  {
+    area = (boundingBoxes.row(2) - boundingBoxes.row(0)) %
+        (boundingBoxes.row(3) - boundingBoxes.row(1));
+  }
+  else
+  {
+    area = (boundingBoxes.row(2)) % (boundingBoxes.row(3));
+  }
+
+  while (sortedIndices.n_elem > 0)
+  {
+    size_t selectedIndex = sortedIndices(sortedIndices.n_elem - 1);
+
+    // Choose the box with the largest probability.
+    selectedBoundingBoxes.insert_cols(0, boundingBoxes.col(selectedIndex));
+
+    // Check if there are other bounding boxes to compare with.
+    if (sortedIndices.n_elem == 1)
+    {
+      break;
+    }
+
+    // Remove the last index.
+    sortedIndices = sortedIndices(arma::span(0, sortedIndices.n_rows - 2),
+        arma::span());
+
+    // Calculate IoU of remaining boxes with the last bounding box with
+    // the highest confidence score.
+    BoundingBoxesType intersectionArea;
+    if (UseCoordinates)
+    {
+      intersectionArea = arma::clamp(arma::clamp(
+          boundingBoxes.submat(arma::uvec(1).fill(2), sortedIndices), DBL_MIN,
+          boundingBoxes(2, selectedIndex)) - arma::clamp(
+          boundingBoxes.submat(arma::uvec(1).fill(0), sortedIndices),
+          boundingBoxes(0, selectedIndex), DBL_MAX), 0.0, DBL_MAX) %
+          arma::clamp(arma::clamp(boundingBoxes.submat(arma::uvec(1).fill(3),
+          sortedIndices), DBL_MIN, boundingBoxes(3, selectedIndex)) -
+          arma::clamp(boundingBoxes.submat(arma::uvec(1).fill(1),
+          sortedIndices), boundingBoxes(1, selectedIndex), DBL_MAX),
+          0.0, DBL_MAX);
+    }
+    else
+    {
+      intersectionArea = arma::clamp(arma::clamp(
+          boundingBoxes.submat(arma::uvec(1).fill(2), sortedIndices) +
+          boundingBoxes.submat(arma::uvec(1).fill(0), sortedIndices), DBL_MIN,
+          boundingBoxes(2, selectedIndex) + boundingBoxes(0, selectedIndex)) -
+          arma::clamp(boundingBoxes.submat(arma::uvec(1).fill(0),
+          sortedIndices), boundingBoxes(0, selectedIndex), DBL_MAX), 0.0,
+          DBL_MAX) % arma::clamp(arma::clamp(
+          boundingBoxes.submat(arma::uvec(1).fill(3), sortedIndices) +
+          boundingBoxes.submat(arma::uvec(1).fill(1),
+          sortedIndices), DBL_MIN, boundingBoxes(3, selectedIndex) +
+          boundingBoxes(1, selectedIndex)) -
+          arma::clamp(boundingBoxes.submat(arma::uvec(1).fill(1),
+          sortedIndices), boundingBoxes(1, selectedIndex), DBL_MAX),
+          0.0, DBL_MAX);
+    }
+
+    BoundingBoxesType calculateIoU = intersectionArea /
+        (area(sortedIndices).t() - intersectionArea + area(selectedIndex));
+
+    sortedIndices.shed_rows(arma::find(calculateIoU > threshold));
+  }
+}
```

Comment:
Our output, `selectedBoundingBoxes` will have the selected bounding boxes in the ascending order of the `confidenceScores` and I think it's more reasonable to store the the boxes **in the descending order**. By doing that, I think users can more easily access the matrix that is from the largest confidence score.
```suggestion
  }
  selectedBoundingBoxes = arma::fliplr(selectedBoundingBoxes);
}
```

Reply:
That makes sense. Thanks a lot. Making the changes now.
Regards.